nixos/modules/services/monitoring/opnborg.nix

```diff
@@ -0,0 +1,104 @@
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}:
+with lib;
+let
+  cfg = config.services.opnborg;
+in
+{
+  options.services.opnborg = {
+    enable = mkEnableOption "opnborg";
+
+    user = mkOption {
```

Comment:
Do you need a fixed user for this service? Or can a systemd DynamicUser be used?

I think the main reason for needing a "real" user is if a local user needs to interact with the data directly, outside of the service. If the data is only accessed via the service than DynamicUser is the easiest solution.

Reply:
The backup data from the appliances contain sensitive information. 

Having a dedicated user as owner might help develop individual firewall / security concepts if needed. 

When this is a blocking point, I can remove the user. Its no hard requirement within the application itself.

Reply:
It's not about having a separate user. I think that makes sense. It's about creating an explicit user versus letting systemd create one implicitly via [DynamicUser](https://www.freedesktop.org/software/systemd/man/latest/systemd.exec.html#DynamicUser=).

It's explained in more detail when to use what in https://discourse.nixos.org/t/predefined-uids-gids-when-to-use-them-and-when-not-to-use-them/956/3

Reply:
I may need some guidance how to solve the following conflict:

- The application clearly needs a fixed reliable deterministic gid/guid to store/backup/restore the appliances config data (/var/lib/) that contain secrets (keys,...)

- The appliance (open source firewall / infrastructure / cluster) audience is rather small, so reserve systemuser deterministic uid/gid below 399 via nixos/modules/misc/ids.nix might be not justified !?